app/controllers/application_controller.rb

```diff
@@ -27,6 +27,42 @@ def validation_error(resource)
     }, status: :bad_request
   end
 
+  def valid_user?(user_id)
```

Comment:
The way that you handled all the errors in the ApplicationController is great! This allows you not to repeat the code in every controller. But I think that for checking the user, you should use current_user instead of this method.